oxy/poc_translator.py

```diff
@@ -0,0 +1,73 @@
+from ruamel.yaml import YAML
+from functools import reduce
+
+
+yaml: YAML = YAML(typ="safe")
+
+'''
+pub fn decode_accumulator_status(data: &[u8]) -> HashMap<u8, f32> {
+    let mut result = HashMap::new();
+    result.insert(1, (pd::big_endian(&data[0..2], 8) as f32) / 10.0);
+    result.insert(
+        2,
+        pd::twos_comp(pd::big_endian(&data[2..4], 8) as u32, 16) as f32 / 10.0,
+    );
+    result.insert(3, pd::big_endian(&data[4..6], 8) as f32);
+    result.insert(4, data[6] as f32);
+    result.insert(5, data[7] as f32);
+    result
+}
+'''
+out_string: str = ""
+data = yaml.load(open("mapping.yaml"))[1]
+# print(data)
+
+function_name: str = "decode" + "_" + "_".join(data['string'].split(" "))
+args: str = "(data: &[u8])"
+returnVal: str= " -> HashMap<u8, f32>"
+
+signature: str = "pub fn " + function_name + args + returnVal + " {"
+instantiate_hash_map: str = "    let mut result = HashMap::new();"
+conclusion: str = "    result\n}"
```

Comment:
Similar to the comment on using classes to represent the data in the YAML file, I think each component of the rust code would def benefit from putting it into a class.